src/wtforms/validators.py

```diff
@@ -676,6 +680,40 @@ def __call__(self, hostname):
         return True
 
 
+class ReadOnly:
+    """
+    Set a field readonly.
+
+    Validation fails if the form data is different than the
+    field object data, or if unset, from the field default data.
+    """
+
+    def __init__(self):
+        self.field_flags = {"readonly": True}
+
+    def __call__(self, form, field):
+        if field.data != field.object_data:
+            raise ValidationError(field.gettext("This field cannot be edited"))
+
+
+class Disabled:
+    """
+    Set a field disabled.
+
+    Validation fails if the form data is different than the
+    field object data, or if unset, from the field default data.
```

Comment:
That's not correct

Reply:
Fixed. Now this is `Validation fails if the form data has any value.`